### PR TITLE
Update travis config

### PR DESCRIPTION
**Pull request by ismay**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,15 @@
 language: node_js
-cache: yarn
```
Comment (XhmikosR) [inline .travis.yml]: You should cache node modules here.

File: .travis.yml [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
 notifications:
   email: false
 node_js:
   - "4"
+  - "5"
```
Comment (XhmikosR) [inline .travis.yml]: 5 and 7 are useless dev releases and shouldn't be tested...

File: .travis.yml [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
   - "6"
+  - "7"
   - "8"
+install:
+  - npm i -g npm
```
Comment (XhmikosR) [inline .travis.yml]: You shouldn't do this for all node versions; you should do this only where is needed, probably 4 and 6 like this

```yml
before_install:
  - if [[ `npm -v` != 5* ]]; then npm install -g npm@5; fi
```

File: .travis.yml (hunk 1 above)
Comment (XhmikosR) [reply .travis.yml]:
```yml
cache:
  directories:
    - node_modules
```

**Comment by XhmikosR** [comment]
@ismay: just my 2 cents, I hope you don't mind me commenting. :)

File: .travis.yml (hunk 2 above)
Comment (ismay) [reply .travis.yml]: Ah you mean only test the LTS versions. Yeah that makes sense.

File: package-lock.json [hunk 4]
```diff
@@ -4,6 +4,15 @@
   "lockfileVersion": 1,
   "requires": true,
   "dependencies": {
+    "JSONStream": {
```
Comment (XhmikosR) [inline package-lock.json]: Weird you get these changes; I'm using node 8.5.0, and manually cleaned up the `node_modules` folder and did `npm i` to get a fresh `package-lock.json` file. Not a big deal, I just don't get why this happens.

File: .travis.yml (hunk 1 above)
Comment (ismay) [reply .travis.yml]: Ah nice

File: package-lock.json (hunk 4 above)
Comment (ismay) [reply package-lock.json]: Yeah I know. I'm on 8.4.0 and 5.3.0. I've been encountering this on a lot of projects. It's weird.

File: .travis.yml [hunk 5]
```diff
@@ -8,8 +8,9 @@ node_js:
   - "4"
   - "6"
   - "8"
+before_install:
+  - if [[ `npm -v` != 5* ]]; then npm install -g npm@5; fi
 install:
-  - npm i -g npm
   - npm i
```
Comment (XhmikosR) [inline .travis.yml]: You can remove the install step completely; `npm i` is always done.

**Comment by ismay** [comment]
> just my 2 cents, I hope you don't mind me commenting. :)

No definitely not, they're good suggestions :) 👍

File: .travis.yml (hunk 5 above)
Comment (ismay) [reply .travis.yml]: Ah yeah you're right

File: .travis.yml [hunk 6]
```diff
@@ -1,11 +1,15 @@
...
 notifications:
   email: false
 node_js:
   - "4"
   - "6"
   - "8"
+before_install:
+  - if [[ `npm -v` != 5* ]]; then npm install -g npm@5; fi
 script:
   - npm test
```
Comment (XhmikosR) [inline .travis.yml]: We could even remove the script step since `npm test` is the default one. But it's nitpicking. :)

Comment (ismay) [reply .travis.yml]: Well while we're at it :)

**Pull request merged**